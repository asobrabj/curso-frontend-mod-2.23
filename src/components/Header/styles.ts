import styled from 'styled-components'
import { ContainerTheme } from '../ChangeTheme/styles'
import { NewAnotation } from '../NewAnotation/styles'
import { Input } from '../Search/styles'

export const Container = styled.header`
  position: relative;
  display: flex;
  padding: 16px;
  border-bottom: 1px solid ${(props) => props.theme.color.shadow};

  > div {
    display: flex;
    align-items: center;
  }
`

export const Heading = styled.h1`
  margin-left: 16px;
  font-style: italic;
  > a {
    text-decoration: none;
  }
`

export const ContainerExtras = styled.div`
  @media (max-width: 820px) {
    > ${Input} {
      display: none;
    }
    > ${NewAnotation} {
      display: none;
    }
    > ${ContainerTheme} {
      display: none;
    }
  }
`
