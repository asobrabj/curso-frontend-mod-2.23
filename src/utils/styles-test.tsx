import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import theme from '../styles/darkTheme'

export const renderTheme = (Children: React.ReactNode) => {
  return render(<ThemeProvider theme={theme}> {Children} </ThemeProvider>)
}
