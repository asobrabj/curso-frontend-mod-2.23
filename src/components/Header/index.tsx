/* eslint-disable react-hooks/exhaustive-deps */
import { MdOutlineChecklistRtl } from 'react-icons/md'
import ChangeTheme from '../ChangeTheme'
import NewAnotation from '../NewAnotation'
import Search from '../Search'
import * as S from './styles'

interface ITheme {
  isDark: boolean
  changeTheme: () => void
}

export default function Header({ changeTheme, isDark }: ITheme) {
  return (
    <S.Container className="container">
      <div>
        <MdOutlineChecklistRtl />
        <S.Heading>
          {' '}
          <a href="/">Anotações</a>{' '}
        </S.Heading>
      </div>
      <S.ContainerExtras>
        <Search />
        <ChangeTheme changeTheme={changeTheme} isDark={isDark} />
        <NewAnotation />
      </S.ContainerExtras>
    </S.Container>
  )
}
