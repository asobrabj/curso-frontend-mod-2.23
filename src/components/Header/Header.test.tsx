import { fireEvent, screen } from '@testing-library/react'
import Header from '.'
import { renderTheme } from '../../utils/styles-test'

jest.mock('../../context/localStorage/localStorage', () => ({
  useStorage: jest.fn(() => ({
    anotations: [],
    setAnotations: jest.fn(),
    getStorage: jest.fn(),
    getFilters: jest.fn(),
  })),
}))

describe('Header Component', () => {
  test('Renderiza o componente corretamente', () => {
    renderTheme(<Header changeTheme={() => {}} isDark={false} />)

    expect(screen.getByText('Anotações')).toBeInTheDocument()
    expect(screen.getByPlaceholderText('filtrar anotações')).toBeInTheDocument()
    expect(screen.getByTitle('change theme')).toBeInTheDocument()
    expect(screen.getByTitle('new-anotation')).toBeInTheDocument()
  })
  test('Chama a função changeTheme quando o botão de troca de tema é clicado', () => {
    const changeThemeMock = jest.fn()
    renderTheme(<Header changeTheme={changeThemeMock} isDark={false} />)

    fireEvent.click(screen.getByTitle('change theme'))

    expect(changeThemeMock).toHaveBeenCalledTimes(1)
  })
  it('snapshot', () => {
    const { container } = renderTheme(<Header />)
    expect(container).toMatchSnapshot()
  })
})
