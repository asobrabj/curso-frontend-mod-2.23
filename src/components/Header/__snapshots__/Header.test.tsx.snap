// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component snapshot 1`] = `
.c6 {
  cursor: pointer;
  margin-left: 450px;
  margin-top: 5px;
  margin-right: 80px;
  position: relative;
  border: 1px solid #999;
  width: 32px;
  height: 32px;
  border-radius: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
}

.c6 >i {
  position: absolute;
  top: 6px;
  left: 6px;
  font-size: 18px;
  border-radius: 50%;
  color: #bdbbbb;
}

.c8 {
  position: absolute;
  top: 10px;
  right: 70px;
  display: flex;
  align-items: center;
  cursor: pointer;
  padding: 8px;
}

.c9 {
  font-weight: bold;
  margin-right: 8px;
}

.c10 {
  font-size: 24px;
  margin-top: 5px;
}

.c4 {
  margin-left: 120px;
  border: 1px solid;
  border-radius: 8px;
  width: 300px;
  padding: 8px;
  font-size: 16px;
  font-weight: bold;
}

.c4::placeholder {
  color: #999;
}

.c0 {
  position: relative;
  display: flex;
  padding: 16px;
  border-bottom: 1px solid #999;
}

.c0 >div {
  display: flex;
  align-items: center;
}

.c1 {
  margin-left: 16px;
  font-style: italic;
}

.c1 >a {
  text-decoration: none;
}

@media (max-width: 820px) {
  .c6 {
    margin-left: 226px;
  }
}

@media (max-width: 820px) {
  .c2 >.c3 {
    display: none;
  }

  .c2 >.c7 {
    display: none;
  }

  .c2 >.c5 {
    display: none;
  }
}

<div>
   
  <header
    class="c0 container"
  >
    <div>
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M11 7H2v2h9V7zm0 8H2v2h9v-2zm5.34-4L12.8 7.46l1.41-1.41 2.12 2.12 4.24-4.24L22 5.34 16.34 11zm0 8l-3.54-3.54 1.41-1.41 2.12 2.12 4.24-4.24L22 13.34 16.34 19z"
        />
      </svg>
      <h1
        class="c1"
      >
         
        <a
          href="/"
        >
          Anotações
        </a>
         
      </h1>
    </div>
    <div
      class="c2"
    >
      <input
        class="c3 c4"
        name="search"
        placeholder="filtrar anotações"
        type="text"
        value=""
      />
      <div
        class="c5 c6"
        data-testid="change theme"
        title="change theme"
      >
        <i
          title="icon theme"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M264 480A232 232 0 0132 248c0-94 54-178.28 137.61-214.67a16 16 0 0121.06 21.06C181.07 76.43 176 104.66 176 136c0 110.28 89.72 200 200 200 31.34 0 59.57-5.07 81.61-14.67a16 16 0 0121.06 21.06C442.28 426 358 480 264 480z"
            />
          </svg>
        </i>
      </div>
      <div
        class="c7 c8"
        title="new-anotation"
      >
        <p
          class="c9"
        >
          Nova Anotação
        </p>
        <i
          class="c10"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              id="Circle_Plus"
            >
              <g>
                <path
                  d="M15,12.5H12.5V15a.5.5,0,0,1-1,0V12.5H9a.5.5,0,0,1,0-1h2.5V9a.5.5,0,0,1,1,0v2.5H15A.5.5,0,0,1,15,12.5Z"
                />
                <path
                  d="M12,21.932A9.934,9.934,0,1,1,21.932,12,9.944,9.944,0,0,1,12,21.932ZM12,3.065A8.934,8.934,0,1,0,20.932,12,8.944,8.944,0,0,0,12,3.065Z"
                />
              </g>
            </g>
          </svg>
        </i>
      </div>
    </div>
  </header>
   
</div>
`;
